ampl-converter/src/main/java/com/powsybl/ampl/converter/version/AmplExportVersionImpl.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.ampl.converter.version;
+
+/**
+ * @author Nicolas Pierre <nicolas.pierre at artelys.com>
+ */
+public enum AmplExportVersionImpl implements AmplExportVersion {
+    /**
+     * Legacy export
+     */
+    V1_LEGACY(BasicAmplExporter.getFactory());
```

Comment:
Is `V1_LEGACY` a value defined in AMPL or is it a custom value? If it is a PowSyBl value, it would be better to call it simply `V1`.

Reply:
@geofjamg @annetill Do you know what this value "V1_LEGACY" means? (I do not know the current value of versioning for Powsybl or Ampl or...)

Reply:
I added an new commit with the change "V1_LEGACY" -> "1.0", I was not entirely sure of what was used: 
The string AmplExportVersion.V1_0.name() (value "V1_0") is used in the code but I do not think it is used in the exporter ampl files (examples here? ./ampl-converter/src/test/resources/inputs/ ?). In these files it is the string AmplColumnsExporter.EXPORTER_ID which is used (value "1.0")

Reply:
To clarify this, I suggest to add in the enum:
- another attribute `exporterId` (with the value "1.0" for "V1_0");
- a getter to retrieve this attribute;
- a method `fromExporterId` which will take a String and return the corresponding item.

Thus:
- `BasicAmplExporter.EXPORTER_ID` could be removed and `BasicAmplExporter.getExporterId()` will return `V1_0.getExporterId()`;
- `AmplExportVersion.valueOf(...)` usages should be replaced by `fromExporterId(...)`;
- `AmplExportVersion.name()` usages should be replaced by `getExporterId()`.

This way, the user will have to use `1.0` instead of `1_0` in the config file for the `EXPORT_VERSION_PARAMETER` parameter.
<br/>

Could you also add 2 unit tests to check that:
- when `EXPORT_VERSION_PARAMETER` have the value "1.0" the export works;
- when the parameter has an unexpected value, the export fails ?

Reply:
When we speak about config file, is it the Properties "iidm.export.ampl.export-version"?
for example:

```
    @Test
    void writeHeadersWithVersion() throws IOException {
        Network network = Network.create("dummy_network", "test");
        MemDataSource dataSource = new MemDataSource();

        Properties properties = new Properties();
        properties.put("iidm.export.ampl.export-version", "1.0");

        export(network, properties, dataSource);
        assertEqualsToRef(dataSource, "_headers", "inputs/headers.txt");
    }
```

Reply:
@olperr1 AmplExportVersion.V1_0 is given as "default value":
- for constructor(s) AmplExportConfig
- for EXPORT_VERSION_PARAMETER (Parameter.defaultValue)  (now it is AmplExportVersion.V1_0.getExporterId() but it is the same problem) 
Do you think this code will be untouched when we add a new ampl export version (meaning we want to use the "legacy" version here) or it will be updated with the last version (meaning we want to use the current version here)?

Reply:
Yes, sorry I shouldn't have mention the config files. In fact properties can be read from a "config.yml" file (see for instance this [test file](https://github.com/powsybl/powsybl-core/blob/main/cgmes/cgmes-completion/src/test/resources/com/powsybl/config/test/config.yml#L3) and the [corresponding parameter](https://github.com/powsybl/powsybl-core/blob/ff84c4e746eb5f1248286839195620a24bcfd552/cgmes/cgmes-completion/src/main/java/com/powsybl/cgmes/completion/CreateMissingContainersPreProcessor.java#L77)).

Reply:
>       Do you think this code will be untouched when we add a new ampl export version (meaning we want to use the "legacy" version here) or it will be updated with the last version (meaning we want to use the current version here)?
I am not quite sure. But either way, I think it is a good idea to use a method returning the default value.